apps/backend/src/routers/auth.ts

```diff
@@ -0,0 +1,49 @@
+import { procedure, router } from '../trpc';
+import { z } from 'zod';
+import { db } from 'src/db';
+import { RDS } from 'src/lib/rds';
+
+const authRouter = router({
+    handleGuestSession: procedure.input(z.object({ name: z.string() })).query(async ({ input }) => {
+        const account = await RDS.registerUserAccount(db, input.name, input.name, 'GUEST');
+
+        if (account.userId.length > 0) {
+            let session = await RDS.upsertSession(db, account.userId);
+            return session?.refreshToken;
+        }
+        return null;
+    }),
+    /**
+     * Returns the current session, returns true if the session exists exist and hasn't expired
+     */
+    validateSession: procedure.input(z.object({ token: z.string() })).query(async ({ input }) => {
+        if (input.token === '') return false;
+        const session = await RDS.getCurrentSession(db, input.token);
+
+        return session !== null && session.expires > new Date();
+    }),
+    /**
+    Removes a session from the database
+     */
+    removeSession: procedure.input(z.object({ token: z.string() })).mutation(async ({ input }) => {
```

Comment:
Since this is the opposite of `validateSession`, from my understanding, it should be called `invalidateSession`